Unity-Technologies/ScriptableRenderPipeline
# UniversalRP increase light limits

### Purpose of this PR
 Increases light limits for UniversalRP. 

 New limits: 
    Per-object lights: 8, except in GLES2 that is 4.
    Visible Lights: 256 if not mobile, 32 otherwise.

 Non-mobile path store lights in StructuredBuffer.
 Currently there are some performance issues with StructuredBuffer in mobile. It uses UBO path instead.

 Performance Test on Android. No regression found
 There's some improvement in performance in some Android GLES3 devices.

---
### Release Notes
 Added
---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: 
  - Tested UniversalGraphicsTest project in GLES3, Vulkan, Metal and Nintendo Switch. No regressions found.
  - Performance Testing on GLES3 Potato Benchmark. Same or better performance.
  - QA validated template and BoatAttack

- [x ] Opened test project + Run graphic tests locally
- [x ] Built a player
- [x ] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: 
 - Added two new automated tests. 
    008 - Tests up to 8 per-object lights
    009 - Tests directional lights + 32 additional lights casting shadows.

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=lw/increase-light-limits&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: High. Changes how lighting data is setup in UniversalRP.

**Halo Effect**: High. Touches all platforms in different ways.

---
### Comments to reviewers
Notes for the reviewers you have assigned.
